networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -139,6 +174,7 @@ Graph NetworkitBinaryReader::read(const std::string& path) {
             }
             //Read adjacency lists.
             for (uint64_t j = 0; j < outNbrs; j++) {
+                edgeid id;
```

Comment:
Don't we have to set the `Graph`'s maximal edge ID somewhere? Otherwise the `Graph` will allocate the same ID again when adding new edges.

Reply:
No, because we are reading the ids from file, not generating ourselves. See line 213.